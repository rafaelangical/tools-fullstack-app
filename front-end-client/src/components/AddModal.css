.App-modal-add{
  width: 100%;
  height: 100%;
  display: flex;
  flex-flow: row wrap;
  justify-content: center;
  align-items: center;
  position: absolute;
  position: fixed;
  background-color: rgba(1,1,1,0.5); 
  top: 0;
}
.div-modal-add-full {
  width: 50%;
  height: 60%;
  margin: 0 auto;
  background-color: white;
  display: flex;
  flex-flow: row wrap;
}
.div-modal-add-top{
  width: 80%;
  height:  10%;
  margin: 0 auto;
  justify-content: center;
}
.div-modal-add-middle{
  width: 80%;
  height: 70%;
  margin: 0 auto;
  display: flex;
  justify-content: space-around;
  flex-flow: column wrap;
  align-content: space-between;
}
.div-modal-add-middle > input{
  width: 90%;
  height: 6%
}
.div-modal-add-bottom{
  width: 80%;
  height: 10%;
  margin: 0 auto;
  display: flex;
  flex-flow: column wrap;
  align-content: flex-end;
}
textarea{
  width: 90%;
  height: 25%;
}