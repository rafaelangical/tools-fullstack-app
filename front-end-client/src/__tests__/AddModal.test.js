import React from 'react';
import AddModal from '../components/AddModal';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';

Enzyme.configure({ adapter: new Adapter() })

describe('Test if AddModal is rendered correctly', () => {
  test('renders correctly AddModal', () => {
    const tree = renderer.create(
      <AddModal />
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('test a state of AddModal', () => {
  test('have title state', () => {
    const wrapper = shallow(<AddModal />);
    const titleState = wrapper.state('title');
    expect(titleState).toBeDefined();
  });
  
  test('have description state', () => {
    const wrapper = shallow(<AddModal />);
    const descriptionState = wrapper.state('description');
    expect(descriptionState).toBeDefined();
  });

  test('have link state', () => {
    const wrapper = shallow(<AddModal />);
    const linkState = wrapper.state('link');
    expect(linkState).toBeDefined();
  });

  test('have tags state', () => {
    const wrapper = shallow(<AddModal />);
    const tagsState = wrapper.state('tags');
    expect(tagsState).toBeDefined();
  });

});

describe('test setState', () => {
  test('test setState of title', () => {
    const wrapper = mount(<AddModal />);
    wrapper.setState({ title: 'title test' });
  });

  test('test setState of description', () => {
    const wrapper = mount(<AddModal />);
    wrapper.setState({ description: 'description test of setstate' });
  });
  
  test('test setState of link', () => {
    const wrapper = mount(<AddModal />);
    wrapper.setState({ link: 'globo.com' });
  });
  
  test('test setState of tags', () => {
    const wrapper = mount(<AddModal />);
    wrapper.setState({ tags: 'jsjsj' });
  });    
});
