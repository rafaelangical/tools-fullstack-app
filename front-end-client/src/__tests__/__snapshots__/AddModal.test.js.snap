// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if AddModal is rendered correctly renders correctly AddModal 1`] = `null`;
